torch_xla/csrc/tensor.cpp

```diff
@@ -1769,20 +1802,26 @@ std::shared_ptr<XLATensor::Async> XLATensor::SyncTensorsGraphInternal(
     TensorCollectionBarrier(&coll);
     return nullptr;
   }
-  PostOrderData po_data = RunPostOrder(*tensors, &coll);
+  std::vector<torch::lazy::Value> ir_values;
+  std::vector<torch::lazy::BackendDataPtr> tensor_data_vec;
+  FetchTensorData(tensors, coll.config, coll.indices, ir_values,
```

Comment:
I think `FetchTensorData` is no longer an accurate name. Can you rename it to something along the line of `ExtractIRAndPrepareXlaData`

Reply:
Fair point. Let me rename. Thanks!